dvc/command/dag.py

```diff
@@ -72,7 +72,8 @@ def run(self):
         try:
             target = None
             if self.args.target:
-                stages = self.repo.collect(self.args.target)
+                pairs = self.repo.collect_granular(self.args.target)
+                stages = [stage for stage, _ in pairs]
```

Comment:
For `--outs` we really need to not ignore the `filter_info` we get from pairs and possibly only show the filtered outputs in the outs DAG.

Reply:
Fixed.